evaluate_test.go

```diff
@@ -22,22 +22,21 @@ type expressionTest struct {
 var evaluateTests map[string]expressionTest = map[string]expressionTest{
 	"Flat Struct": {
 		testFlatStruct{
-			Int:        -1,
-			Int8:       -2,
-			Int16:      -3,
-			Int32:      -4,
-			Int64:      -5,
-			Uint:       6,
-			Uint8:      7,
-			Uint16:     8,
-			Uint32:     9,
-			Uint64:     10,
-			Float32:    1.1,
-			Float64:    1.2,
-			Bool:       true,
-			String:     "exported",
-			unexported: "unexported",
```

Comment:
Did the unexported field test need to be removed? I am pretty sure it should still result in an error as pointer structure will be ignoring unexported fields.

Reply:
Pointerstructure errors on unexported fields.

Reply:
Actually I think I took it out because pointerstructure _panics_ on unexported fields. I could do a recover if you want.

Reply:
It shouldn't panic as unexported fields are specifically being ignored.

https://github.com/mitchellh/pointerstructure/blob/a15bd03913edf1aab7320050de0cfdd654679269/get.go#L113-L116

Which will result in an error about the struct field not being present. I guess I could have made the error more specific there.

Also just verified that with an extra test in pointerstructure.

So no recover is necessary. Its probably fine to leave the test out as this is now pointerstructure behavior though.

Reply:
Weird, I had some reason I decided to remove this rather than change the error message. I thought it was because it was panicking. I may add it back in, perhaps it was panicking in go-bexpr _after_ getting the value back from pointerstructure (maybe an error was being ignored or something). 

I'm going to unresolve this just to remind me to check this again.

Reply:
👍

Reply:
Just tested. It seems fine, just updated the error. My guess is that I was seeing issues in this part: https://github.com/hashicorp/go-bexpr/pull/14/files/0733d7a5405945d82019a853191d806ff2365b10#diff-c1d420d0ae392cdada4d259285179d198b39ec5723246161ef8297bb242d7c8fL232 prior to simply removing all of that very custom stuff, so when I removed the unexported fields there I simply removed them everywhere. I've added them back in to the struct types though.